### PR TITLE
Handle missing data from the API

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following ensures that we handle missing data returned back
from the new API. Although the new API isn't turned on in this
branch, we should at least handle the case where it could be.

### CODE REVIEW

File: juju/bundle.py [hunk 1]
```diff
@@ -681,9 +681,12 @@ def method():
             await context.model.create_offer(ep, offer_name=self.offer_name, application_name=application)
 
     def __str__(self):
+        endpoints = ""
+        if self.endpoints is not None:
+            endpoints = self.endpoints.join(",")
         return "create offer {offer_name} using {application}:{endpoints}".format(offer_name=self.offer_name,
                                                                                   application=self.application,
-                                                                                  endpoints=self.endpoints.join(","))
+                                                                                  endpoints=endpoints)
```
Comment (achilleasa) [inline juju/bundle.py]: Small nitpick here. If endpoints is empty the returned string will end with a colon. How about if you changed L686 to read `":" + self.endpoints.join(",")` and remove the colon from L687?

Comment (SimonRichardson) [reply juju/bundle.py]: Updated and fixed, good spot.

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by mitechie** [comment]
$$merge$$

**Pull request merged**